crates/iota-json-rpc-api/src/extended.rs

```diff
@@ -79,4 +79,19 @@ pub trait ExtendedApi {
     /// Exclusively served by the indexer.
     #[method(name = "getParticipationMetrics")]
     async fn get_participation_metrics(&self) -> RpcResult<ParticipationMetrics>;
+
+    /// Return list of transactions for a specified query criteria.
+    #[rustfmt::skip]
+    #[method(name = "queryTransactionBlocks")]
+    async fn query_transaction_blocks_v2(
+        &self,
+        /// the transaction query criteria.
+        query: IotaTransactionBlockResponseQueryV2,
+        /// An optional paging cursor. If provided, the query will start from the next item after the specified cursor. Default to start from the first item if not specified.
+        cursor: Option<TransactionDigest>,
+        /// Maximum item returned per page, default to QUERY_MAX_RESULT_LIMIT if not specified.
+        limit: Option<usize>,
+        /// query result ordering, default to false (ascending order), oldest record first.
+        descending_order: Option<bool>,
+    ) -> RpcResult<TransactionBlocksPage>;
```

Comment:
@samuel-rufi just realized that this is part of the `ExtendedApi`. There is no reason to separate it. We can't version it properly like this and it is the same method. You can still add it in the `IndexerApi` and implement that in the node as well, throwing `UnsupportedFilter` error in case the new filter is used.